components/definition/_definition.scss

```diff
@@ -0,0 +1,36 @@
+.definition {
+  $block: &;
+
+  #{$block}__link {
```

Comment:
Not a fan of this level of Scss syntax complexity. It's makes the code really hard to understand what's going on if you're not super familiar with it.

Reply:
Fundamentally disagree with this. The code is no more complex than what you would find in other places in the project, some mixins are very complex. This code is concise and DRY and again reinforces BEM policies using naming conventions for variable $block. This reinforces nesting to only be one level deep. There are countless examples in this repository of code that breaks BEM rules already making it hard to compose components.

Reply:
I think there's a balance between using BEM as a convention and enforcing a rigid structure. For example, you can enforce nesting rules using peer review/and or linting.

Agreed, there are some complex mixins but they aren't maintained day-to-day whereas writing components in this style makes the majority of the code more complex and difficult to understand. This potentially puts up a barrier to designers who know a bit of CSS contributing to the pattern library as well as on-boarding less experienced devs.

Reply:
I wouldn't say this code is rigid, its modular and highly extensible but also serves as an example which clearly defines how BEM is meant to be used. Having said that, to help with readability in the definition class, #{$block} can be replaced with the hard-coded 'definition'. This, I don't think would interfere with how the block class is defined or affect its maintainability.

Unfortunately as a separate issue, all current code in the project has been reviewed but still a lot of problems have passed through, fixing these problems will be expensive.

I think the main problem around complexity appears to be down to understanding syntax--the project requires front end developers to maintain its code-base, not designers. Designers are consumers of the pattern library output and for them nothing is different, the CSS will look the same. Front end developers as maintainers of the pattern library are responsible for writing the source code, they should be able to use tried and tested techniques to maintain healthy high-quality and decoupled source code. This will also help maintainers reduce the complexity in inevitable changes to requirements (this is why styles related to the details-toggle module behaviour are refactored into their own class).

Reply:
I think the main issue is the awkward syntax of `@at-root`. Writing the classes out explicitly is preferable, imo. Although it's more verbose, it's much more readable.

I don't have an issue with using `@extend` so long as the classes extended are placeholders.